staging/src/k8s.io/apiserver/pkg/server/filters/compression.go

```diff
@@ -58,6 +58,18 @@ func WithCompression(handler http.Handler, ctxMapper request.RequestContextMappe
 				return
 			}
 			compressionWriter.Header().Set("Content-Encoding", encoding)
+
+			// periodically flush writer
+			go func() {
```

Comment:
this writer isn't meant to be used from multiple goroutines. spawning a goroutine to flush here introduces races. Flushing on write is safer, if less efficient.

Reply:
done